### DISCUSSION TITLE
Add Local Storage

### DISCUSSION THREAD

**Discussion by Diaapi** [AvaloniaUI/Avalonia#17856] [Ideas]
I would like to see Local Storage in AvaloniaUI and other native APIs like geolocation or the use of the lamp

**Comment by thevortexcloud** [comment]
You can use Maui Essentials/Xamarin essentials for most of that sort of stuff with Avalonia. Avalonia does however have an official API for dealing with file pickers and what not.

**Comment by Diaapi** [comment]
But it wouldn't work on Wasm

**Reply by thevortexcloud** [reply]
You should be able to add the bindings yourself for WASM.

https://learn.microsoft.com/en-us/aspnet/core/client-side/dotnet-interop/?view=aspnetcore-9.0

https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API

**Comment by robloo** [comment]
Note that Avalonia UI focuses only on being a UI framework right now. So this is out of scope (for things like Geolocation or Lamp, etc.).

**Reply by maxkatz6** [reply]
Yes, `storage` is something we support to a degree. But stuff like geolocation/lamp are not considered to be supported.
There are solutions for these already, even though, most of the time, these solutions are not universal/cross-platform.
